pkg/kubectlcmd/client.go

```diff
@@ -47,92 +47,171 @@ func (console) Run(c *exec.Cmd) error {
 	return c.Run()
 }
 
-// Apply runs the kubectl apply with the provided manifest argument
-func (c *Client) Apply(dryRun, verbose bool, kubeconfig, context, namespace string, manifest string, extraArgs ...string) (string, string, error) {
+// kubectlParams is a set of params passed to kubectl.
+type kubectlParams struct {
+	dryRun     bool
+	verbose    bool
+	kubeconfig string
+	context    string
+	namespace  string
+	stdin      string
+	output     string
+	extraArgs  []string
+}
+
+// Apply runs the `kubectl apply` command with parameters:
+// dryRun to not actually run the cmd, verbose to not print manifest
+// kubeconfig, context to identify a cluster,
+// namespace to locate a k8s namespace, and
+// returns stdout, stderr, error getting from running this `kubectl` command.
+func (c *Client) Apply(dryRun, verbose bool, kubeconfig, context, namespace string,
+	manifest string, extraArgs ...string) (string, string, error) {
 	if strings.TrimSpace(manifest) == "" {
-		log.Infof("Empty manifest, not applying.")
+		log.Infof("Empty manifest, not running kubectl apply.")
 		return "", "", nil
 	}
-
-	args := []string{"apply"}
-	if kubeconfig != "" {
-		args = append(args, "--kubeconfig", kubeconfig)
-	}
-	if context != "" {
-		args = append(args, "--context", context)
-	}
-	if namespace != "" {
-		args = append(args, "-n", namespace)
+	subcmds := []string{"apply"}
+	params := &kubectlParams{
+		dryRun:     dryRun,
+		verbose:    verbose,
+		kubeconfig: kubeconfig,
+		context:    context,
+		namespace:  namespace,
+		stdin:      manifest,
+		output:     "",
+		extraArgs:  extraArgs,
 	}
-	args = append(args, extraArgs...)
-	args = append(args, "-f", "-")
-
-	cmd := exec.Command("kubectl", args...)
-	cmd.Stdin = strings.NewReader(manifest)
-
-	var stdout bytes.Buffer
-	var stderr bytes.Buffer
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-
-	cmdStr := strings.Join(cmd.Args, " ")
-	if verbose {
+	return c.kubectl(subcmds, params)
+}
 
-		cmdStr += "\n" + manifest
-	} else {
-		cmdStr += " <use --verbose to see manifest string> \n"
-	}
-	if dryRun {
-		logAndPrint("Apply is in dry run mode, would be applying the following in namespace %s:\n%s\n", namespace, cmdStr)
+// Delete runs the `kubectl delete` command with parameters:
+// dryRun to not actually run the cmd, verbose to not print manifest
+// kubeconfig, context to identify a cluster,
+// namespace to locate a k8s namespace, and
+// returns stdout, stderr, error getting from running this `kubectl` command.
```

Comment:
sorry to nitpick on comments but could you please reformat the descriptions a bit:
```text
// Delete runs the `kubectl delete` command with the following parameters:
￼// dryRun - display the command but don't run it
// verbose - dump the full manifest
￼// kubeconfig, context - used to identify the cluster
￼￼// namespace - k8s namespace for kubectl command
//
// It returns stdout, stderr from the `kubectl` command as strings, and error for errors external to kubectl.
```

Reply:
Fixed.